js_modules/dagster-ui/packages/ui-core/src/assets/auto-materialization/AutomaterializationRoot.tsx

```diff
@@ -0,0 +1,207 @@
+import {gql, useQuery} from '@apollo/client';
+import {
+  Alert,
+  Box,
+  Page,
+  Colors,
+  Checkbox,
+  Spinner,
+  Subtitle2,
+  Heading,
+  PageHeader,
+  Table,
+  ButtonGroup,
+} from '@dagster-io/ui-components';
+import React from 'react';
+
+import {useConfirmation} from '../../app/CustomConfirmationProvider';
+import {useUnscopedPermissions} from '../../app/Permissions';
+import {PYTHON_ERROR_FRAGMENT} from '../../app/PythonErrorFragment';
+import {useQueryRefreshAtInterval} from '../../app/QueryRefresh';
+import {useTrackPageView} from '../../app/analytics';
+import {LiveTickTimeline} from '../../instigation/LiveTickTimeline2';
+import {OverviewTabs} from '../../overview/OverviewTabs';
+import {useAutomaterializeDaemonStatus} from '../AutomaterializeDaemonStatusTag';
+
+import {AutomaterializationEvaluationHistoryTable} from './AutomaterializationEvaluationHistoryTable';
+import {AutomaterializationTickDetailDialog} from './AutomaterializationTickDetailDialog';
+import {AutomaterializeRunHistoryTable} from './AutomaterializeRunHistoryTable';
+import {
+  AssetDameonTicksQuery,
+  AssetDameonTicksQueryVariables,
+  AssetDaemonTickFragment,
+} from './types/AutomaterializationRoot.types';
+
+export const AutomaterializationRoot = () => {
+  useTrackPageView();
+  const automaterialize = useAutomaterializeDaemonStatus();
+  const confirm = useConfirmation();
+
+  const {permissions: {canToggleAutoMaterialize} = {}} = useUnscopedPermissions();
+
+  const queryResult = useQuery<AssetDameonTicksQuery, AssetDameonTicksQueryVariables>(
+    ASSET_DAMEON_TICKS_QUERY,
+  );
+  const [isPaused, setIsPaused] = React.useState(false);
+  useQueryRefreshAtInterval(queryResult, isPaused ? Infinity : 2 * 1000);
+
+  const [selectedTick, setSelectedTick] = React.useState<AssetDaemonTickFragment | null>(null);
+
+  const [tableView, setTableView] = React.useState<'evaluations' | 'runs'>('evaluations');
+
+  const ids = queryResult.data
+    ? queryResult.data.autoMaterializeTicks.map((tick) => `${tick.id}:${tick.status}`)
+    : [];
+  while (ids.length < 100) {
+    // Super hacky but we need to keep the memo args length the same...
+    // And the memo below prevents us from changing the ticks reference every second
+    // which avoids a bunch of re-rendering
+    ids.push('');
+  }
+  const ticks = React.useMemo(
+    () => {
+      return queryResult.data?.autoMaterializeTicks ?? [];
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [...ids.slice(0, 100)],
+  );
```

Comment:
we fetch a new tick array every second via polling which results in new tick references even if the ticks did not change. This memo keeps the tick array the same if nothing actually changed. I could `JSON.stringify` to compare but I thought that would be slower because the objects were huge especially with the decision rules. What we really care about are the id's and the status of each tick, if the status doesn't change then none of the attributes change (they only change when going from in progress -> complete) and if the id's are the same then there's no reason to create a new ticks array with new references which get propagated around and cause everything to re-render.

I used 100 because its a big enough number to make sure every tick in the timeline is included and I need to make sure the array is of size 100 because the number of ticks returned isn't fixed, it depends on how many ticks they've had (initially it'll be 0 and slowly go up) and we need the number of arguments to the useMemo dependency array to be fixed otherwise React complains